src/Detector.php

```diff
@@ -89,6 +89,10 @@ private function take(string $bstr, int $len): array
     {
         $out = [];
         for ($i = 0; $i < $len; $i++) {
+            if (empty($bstr[$i])) {
+                throw new UnreadableStreamException('The stream in offset ' . $i . ' cannot be readable.');
```

Comment:
Could you change `cannot be readable` to `is not readable`? "Readable" is an adjective but "cannot be" would expect a verb after to be grammatically correct. "is not readable" is one easy way to fix this:

>_The stream in offset $i is not readable_